### PR TITLE
Implement RFC8441 Extended CONNECT

### PR DESCRIPTION
**Pull request by ehaydenr**
addresses #92

### CODE REVIEW

**Comment by Lukasa** [comment]
@swift-server-bot add to allowlist

File: Sources/NIOHTTP2/ConnectionStateMachine/HasExtendedConnectSettings.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2024 Apple Inc. and the SwiftNIO project authors
```
Comment (Lukasa) [inline Sources/NIOHTTP2/ConnectionStateMachine/HasExtendedConnectSettings.swift:5]:
```suggestion
// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
```

Comment (Lukasa) [reply Sources/NIOHTTP2/ConnectionStateMachine/HasExtendedConnectSettings.swift:5]: Also, you'll need to update this line:

https://github.com/apple/swift-nio-http2/blob/8d8eb609929aee75336a0a3d2417280786265868/scripts/soundness.sh#L21

It needs to become:

```
sed -e 's/20[12][7890123]-20[12][8901234]/YEARS/' -e 's/20[12][8901234]/YEARS/' 
```

**Review by glbrntt — CHANGES_REQUESTED** [review]
I left a couple of nits but otherwise this looks really good!

File: Sources/NIOHTTP2/HTTP2Error.swift [hunk 2]
```diff
@@ -1111,6 +1115,45 @@ public enum NIOHTTP2Errors {
         }
     }
 
+    /// An unsupported pseudo-header was received.
+    public struct UnsupportedPseudoHeader: NIOHTTP2Error, CustomStringConvertible, @unchecked Sendable {
+        // @unchecked Sendable because access is controlled by getters and copy-on-write setters giving this value semantics
+
+        private var storage: StringAndLocationStorage
```
[the remark this answers is not in the document]
Comment (Lukasa) [reply Sources/NIOHTTP2/HTTP2Error.swift:1122]: That's probably true, but on the other hand maybe we should just schedule a task to fix them all in one go.

[the remark this answers is not in the document]
Comment (glbrntt) [reply Sources/NIOHTTP2/HTTP2Error.swift:1122]: Okay. Let's leave this one as is and sort them all out separately.

**Review by glbrntt — APPROVED** [review]
Nice patch, thanks @ehaydenr!

**Pull request merged**